src/relay/pass/type_infer.cc

```diff
@@ -61,6 +61,22 @@ TVM_REGISTER_API("tvm.relay.type_relation.TupleGetItem")
 .set_body_typed<bool(const Array<Type>&, int, const Attrs&, const TypeReporter&)>(
     TupleGetItemRel);
 
+struct ResolvedTypeInfo {
```

Comment:
I feel we should only allow one constructor, and ask the user to explicitly fill this in